docs/source/tutorials/stable_diffusion.mdx

```diff
@@ -0,0 +1,351 @@
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-->
+
+# Generate images with Stable Diffusion models on AWS Inferentia
+
+## Stable Diffusion
+
+🤗 `Optimum` extends `Diffusers` to support inference on the second generation of Neuron devices(powering Trainium and Inferentia 2). It aims at inheriting the ease of Diffusers on Neuron.
+
+To get started, make sure you have [configured your inf2 / trn1 instance](../installation), and installed optimum:
+
+```bash
+pip install "optimum[neuronx, diffusers]"
+```
+
+### Compile Stable Diffusion
+
+To deploy models, you will need to compile them to TorchScript optimized for AWS Neuron. In the case of Stable Diffusion, there are four components which need to be exported to the `.neuron` format to boost the performance:
+
+* Text encoder
+* U-Net
+* VAE encoder
+* VAE decoder
+
+You can either compile and export a Stable Diffusion Checkpoint via CLI or `NeuronStableDiffusionPipeline` class.
+
+***Export via CLI***
+
+Here is an example of exporting stable diffusion components with `Optimum` CLI:
+
+```bash
+optimum-cli export neuron --model stabilityai/stable-diffusion-2-1-base \
+  --task stable-diffusion \
+  --batch_size 1 \
+  --height 512 `# height in pixels of generated image, eg. 512, 768` \
+  --width 512 `# width in pixels of generated image, eg. 512, 768` \
+  --num_images_per_prompt 4 `# number of images to generate per prompt, defaults to 1` \
+  --auto_cast matmul `# cast only matrix multiplication operations` \
+  --auto_cast_type bf16 `# cast operations from FP32 to BF16` \
+  sd_neuron/
+```
+
+***Export via APIs***
+
+Here is an example of exporting stable diffusion components with `NeuronStableDiffusionPipeline`:
+
+<Tip>
+
+To apply optimized compute of Unet's attention score, please configure your environment variable with `export NEURON_FUSE_SOFTMAX=1`.
+
+Besides, don't hesitate to tweak the compilation configuration to find the best tradeoff between performance v.s accuracy in your use case. By default, we suggest casting FP32 matrix multiplication operations to BF16 which offers good performance with moderate sacrifice of the accuracy. Check out the guide from [AWS Neuron documentation](https://awsdocs-neuron.readthedocs-hosted.com/en/latest/general/appnotes/neuronx-cc/neuronx-cc-training-mixed-precision.html#neuronx-cc-training-mixed-precision) to better understand the options for your compilation.
+
+</Tip>
+
+```python
+>>> from optimum.neuron import NeuronStableDiffusionPipeline
+
+>>> model_id = "runwayml/stable-diffusion-v1-5"
+>>> compiler_args = {"auto_cast": "matmul", "auto_cast_type": "bf16"}
+>>> input_shapes = {"batch_size": 1, "height": 512, "width": 512}
+
+>>> stable_diffusion = NeuronStableDiffusionPipeline.from_pretrained(model_id, export=True, **compiler_args, **input_shapes, device_ids=[0, 1])
```

Comment:
Maybe explain the device_ids here. This is to allow the deployment of multiple models on the same host ?

Reply:
This means in which neuron cores we want to load compiled UNet. If we are using an instance with more than one neuron device, it will be particularly useful, otherwise we just load to both cores of the neuron device by default. Here it's not necessary anymore(I will remove device_ids after the next release as we set the default value when it's not given #297 ).